src/controller/resolvers/userResolver.js

```
@@ -3,6 +3,26 @@ import jsonwebtoken from "jsonwebtoken";
 import { ROLES } from "../../lib/constants";
 
 const resolver = {
+    Query: {
+        user: async (parent, args, context, info) => {
+            // Return error if non-admin tries to access another user's info
+            if (args.id != context.session.user.id  && !context.session.user.role.includes(ROLES.ADMIN)) {
+                throw new Error("User " + context.session.user.id + " attempting to access user ID: " + args.id);
+            }
+
+            const userRecord = await context.db.user.findUnique({
+                where: {
+                    id: args.id
+                }
+            });
+            // Return error if user not found
```

probably not necessary to throw an error here, just return null if the graphql schema allows it (which I just checked - it does), but up to you

---

We already throw an error if the `args.id` doesn't match the user's id (from the request auth header), so if the user isn't found, that's most likely an internal error rather than a user input error (since we know the id and user should exist), so it's worth logging/throwing an error imo so we know if something has gone wrong

---

hmm, I guess we can check this in for now,

still not entirely sure whether we should throw an error, or just return null if they requested a different user, we can go with the error for now, and see if it becomes a problem...

---

Yeah that was one of the points of discussion I had in mind.

I guess the questions to ask are:
1. Are we allowing users to access others' information through this endpoint, or are we creating a different endpoint for that (I feel we'll most likely need one at some point)?
2. If we're not allowing users to access others' information through this endpoint, how do we handle that case?

You've obviously already given an answer for the second point but it can still be open to discussion.